### PR TITLE
OSX build

### PR DESCRIPTION
**Pull request by dfornika**
Addresses #3.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -13,16 +13,17 @@ source:
...
-  skip: True  # [osx or win]
+  skip: True  # [win]
   number: 0
   features:
     - blas_{{ variant }}
 
 requirements:
   build:
     - gcc  # [linux]
+    - llvm  # [osx]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Maybe we should try without this. Though I don't know if we are running into time limit issues.

**Comment by jakirkham** [comment]
It seems like 10.7 and 10.9 are getting mixed up as the minimum macOS version requirement. Should be 10.9 as the minimum.

File: recipe/build.sh [hunk 2]
```diff
@@ -4,6 +4,7 @@ export C_INCLUDE_PATH=${PREFIX}/include
 export LD_LIBRARY_PATH=${PREFIX}/lib
 export LIBRARY_PATH=${PREFIX}/lib
 export CMAKE_PREFIX_PATH=${PREFIX}
+export MACOSX_DEPLOYMENT_TARGET=10.9
```
Comment (jakirkham) [inline recipe/build.sh]: This should already be done by the `toolchain`. We also set this specific value in `conda-forge-build-setup`, which `conda-build` passes through. So I don't think we should need this. Though doesn't hurt to try. Am still mulling over why 10.7 is showing up in the first place.

Comment (dfornika) [reply recipe/build.sh]: Doesn't seem to affect the build:

```
...
pcre2-10.21 configuration summary:
    Install prefix .................. : /Users/travis/miniconda3/conda-bld/julia_1505262068830/work/julia-0.5.2/usr
    C preprocessor .................. : clang -stdlib=libc++ -mmacosx-version-min=10.7 -march=core2 -integrated-as -m64  -E
    C compiler ...................... : clang -stdlib=libc++ -mmacosx-version-min=10.7 -march=core2 -integrated-as -m64 
    Linker .......................... : /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
    C preprocessor flags ............ : 
    C compiler flags ................ : -D_THREAD_SAFE -pthread  -arch x86_64 -mmacosx-version-min=10.9 -m64 -O3 -fvisibility=hidden
    Linker flags .................... :  -arch x86_64 -headerpad_max_install_names -mmacosx-version-min=10.9 -lc++ -Wl,-rpath,'@loader_path/'
    Extra libraries ................. :  
    Build 8-bit pcre2 library ....... : yes
    Build 16-bit pcre2 library ...... : no
...
```

**Comment by jakirkham** [comment]
Have fixed the merge conflict here and pushed a re-rendering to enable macOS builds in this branch (as they are disabled in the feedstock now). Hope this is ok. Just trying to cutdown on Travis CI builds in the org when they are not needed.

**Comment by jakirkham** [comment]
Looking at this more closely, I think the issue is building `pcre2`. So if we can build the `pcre2` package ourselves and get Julia to use it, that build issue should go away. Given this, let's prioritize moving to our `pcre2` and building it for macOS.

**Comment by jakirkham** [comment]
Toggling for CIs.

**Comment by jakirkham** [comment]
Seems to run into issues building `openspecfun` due to the lack of a Fortran compiler. Though am breaking out `openspecfun` as a new package in PR ( https://github.com/conda-forge/staged-recipes/pull/3978 ). FWICT that is the last Fortran-based dependency that Julia has. So we should be able to avoid using a Fortran compiler here entirely (even on Linux).

**Comment by jakirkham** [comment]
Trying to add `openspecfun` as a dependency in PR ( https://github.com/conda-forge/julia-feedstock/pull/26 ). As there are macOS and Linux builds of `openspecfun`, we should be able to easily use it here as well.

File: recipe/build.sh [hunk 3]
```diff
@@ -4,6 +4,7 @@ export C_INCLUDE_PATH=${PREFIX}/include
...
 
 make -j 4 prefix=${PREFIX} MARCH=core2 sysconfigdir=${PREFIX}/etc NO_GIT=1 \
  LIBBLAS=-lopenblas LIBBLASNAME=libopenblas.so LIBLAPACK=-lopenblas LIBLAPACKNAME=libopenblas.so \
```
Comment (jakirkham) [inline recipe/build.sh]: Please change `.so` to `${SHLIB_EXT}`.

File: recipe/build.sh (hunk 2 above)
Comment (jakirkham) [reply recipe/build.sh]: We can drop this line now.

**Comment by jakirkham** [comment]
So we are running into the time limit on Travis CI, which is hardly surprising. That said, it did complete the build. Just not all of the test will be able to run on Travis CI within the time limit. So we may need to come up with a sufficient subset of tests that we run on Travis CI instead.

**Comment by jakirkham** [comment]
Have tried to shorten the number of tests run on macOS to hopefully get them within the time limit. We can revisit adding more tests on macOS if we are able to shorten their run time.

**Comment by jakirkham** [comment]
Weirdly something seems to being getting messed up in the configuration stage of LLVM. Really not following how this is happening given the nature of the last change had nothing to do with the build. Have restarted a few times, but am still getting the same result.

**Comment by dfornika** [comment]
Something to do with this?

https://www.traviscistatus.com/incidents/h6gwvyq9zky0

**Comment by jakirkham** [comment]
So I thought that was impacting how many builds were being run. Though maybe you are right that the cause is deeper.

Guess we can wait until the issue is resolved. Would expect that the last change will pass, but am fine with waiting for CI.

**Comment by jakirkham** [comment]
AFAICT it appears this is because it fails to fetch `libosxunwind`. Looks like issue ( https://github.com/JuliaLang/libosxunwind/issues/5 ) is relevant, which was moved from issue ( https://github.com/JuliaLang/julia/issues/15255 ). I'm not sure how this didn't cause us trouble sooner. One solution would be not to build in parallel. 😒 The other solution would be to build `libosxunwind` separately and make it a dependency. My vote is for the latter. Open issue ( https://github.com/conda-forge/staged-recipes/issues/4015 ) on the topic.

**Comment by abhi18av** [comment]
@dfornika I notice that the linux build works fine now - could you tell me about the trouble you are facing with this recipe ?

**Comment by jakirkham** [comment]
There is some sort of test failure in `backtrace`. Also we cannot run all of the tests as the Travis CI worker will timeout. So have been cherry-pick what I have deemed important tests here given that constraint.

**Comment by dfornika** [comment]
Here is the travis error message:

```
 * backtrace Error During Test
 Test threw an exception of type MethodError
 Expression: length(lkup) == 2
 MethodError: no method matching length(::Void)
 Closest candidates are:
 length(!Matched::SimpleVector) at essentials.jl:168
 length(!Matched::Base.MethodList) at reflection.jl:256
 length(!Matched::MethodTable) at reflection.jl:322
 ...
 in include_from_node1(::String) at ./loading.jl:488
 in include_from_node1(::String) at /Users/travis/miniconda3/conda-bld/julia_1506104291288/_t_env/lib/julia/sys.dylib:?
ERROR: LoadError: LoadError: There was an error during testing
while loading /Users/travis/miniconda3/conda-bld/julia_1506104291288/_t_env/share/julia/test/backtrace.jl, in expression starting on line 40
 in sync_end() at ./task.jl:311
while loading /Users/travis/miniconda3/conda-bld/julia_1506104291288/_t_env/share/julia/test/runtests.jl, in expression starting on line 26
ERROR: A test has failed. Please submit a bug report (https://github.com/JuliaLang/julia/issues)
including error messages above and the output of versioninfo():
Julia Version 0.5.2
Platform Info:
 OS: macOS (x86_64-apple-darwin14.5.0)
 CPU: Intel(R) Xeon(R) CPU X5570 @ 2.93GHz
 WORD_SIZE: 64
 BLAS: libopenblas (DYNAMIC_ARCH NO_AFFINITY Nehalem)
 LAPACK: libopenblas.dylib
 LIBM: libopenlibm
 LLVM: libLLVM-3.7.1 (ORCJIT, nehalem)
TESTS FAILED: julia-0.5.2-blas_openblas_200.tar.bz2
```

The error is triggered on line 40 of the `backtrace` test, which we can inspect [here](https://github.com/JuliaLang/julia/blob/f4c6c9d4bbbd9587d84494e314f692c15ff1f9c0/test/backtrace.jl#L40).

**Comment by dfornika** [comment]
@tkelman @jakirkham @SylvainCorlay

I'd appreciate any advice or ideas ideas on how to push forward with an OS X build . This is beyond my abilities at this point.

**Comment by bgruening** [comment]
Would be nice to get this working for the 1.0 release.

**Comment by jakirkham** [comment]
Agreed. It may be worth trying this on CircleCI. Basically would need a change like [this]( https://github.com/conda-forge/openblas-feedstock/commit/109efcd134512988460b21586680e4a057517fa1 ) and a re-render. It's worth noting that we have had some test failures on macOS that would need to be sorted out as well.

**Comment by jakirkham** [comment]
Closing in favor of PR ( https://github.com/conda-forge/julia-feedstock/pull/55 ).

**Pull request closed without merging**